x/pool-incentives/keeper/grpc_query_test.go

```diff
@@ -138,152 +160,136 @@ func (suite *KeeperTestSuite) TestLockableDurations() {
 		suite.Ctx, notPerpetual, sdk.AccAddress{}, sdk.Coins{}, lockuptypes.QueryCondition{
 			LockQueryType: lockuptypes.ByDuration,
 			Denom:         "stake",
 			Duration:      time.Hour,
 		}, time.Now(), 1)
 
-	suite.NoError(err)
-	err = keeper.UpdateDistrRecords(suite.Ctx, types.DistrRecord{
-		GaugeId: gauge6Id,
-		Weight:  sdk.NewInt(100),
-	})
-	suite.Error(err)
+	suite.Require().NoError(err)
+
+	for _, tc := range []struct {
+		desc                 string
+		poolCreated          bool
+		distRecords          []types.DistrRecord
+		expectedErr          bool
+		expectedRecordLength int
+	}{
+		{
+			desc:                 "No pool exist",
+			poolCreated:          false,
+			distRecords:          []types.DistrRecord{},
+			expectedErr:          false,
+			expectedRecordLength: 0,
+		},
+		{
+			desc:                 "Normal case",
+			poolCreated:          true,
+			distRecords:          []types.DistrRecord{},
+			expectedErr:          false,
+			expectedRecordLength: 3,
+		},
+		{
+			desc:        "Perpetual",
+			poolCreated: true,
+			distRecords: []types.DistrRecord{{
+				GaugeId: gaugePerpetualId,
+				Weight:  sdk.NewInt(300),
+			}},
+			expectedErr:          false,
+			expectedRecordLength: 3,
+		},
+		{
+			desc:        "Non Perpetual",
+			poolCreated: true,
+			distRecords: []types.DistrRecord{{
+				GaugeId: gaugeNonPerpetualId,
+				Weight:  sdk.NewInt(100),
+			}},
+			expectedErr:          true,
+			expectedRecordLength: 0,
+		},
+	} {
+		tc := tc
+		suite.Run(tc.desc, func() {
+			var lockableDurations []time.Duration
+			if tc.poolCreated {
+				// only create a pool
+				pool, err := ammKeeper.GetPoolsAndPoke(suite.Ctx)
+				if len(pool) == 0 {
+					poolId = suite.PrepareBalancerPool()
+				}
+				// LockableDurations should be 1, 3, 7 hours from the default genesis state.
+				lockableDurations = keeper.GetLockableDurations(suite.Ctx)
+				suite.Require().Equal(3, len(lockableDurations))
+
+				var distRecords []types.DistrRecord
+				for i := 0; i < len(lockableDurations); i++ {
+					gaugeId, err := keeper.GetPoolGaugeId(suite.Ctx, poolId, lockableDurations[i])
+					suite.Require().NoError(err)
+
+					distRecords = append(distRecords, types.DistrRecord{
+						GaugeId: gaugeId,
+						Weight:  sdk.NewInt(100),
+					})
+				}
+				tc.distRecords = append(distRecords, tc.distRecords...)
+
+				// Sort in ascending order of gaugeId
+				distRecords = tc.distRecords
+				sort.Slice(distRecords[:], func(i, j int) bool {
+					return distRecords[i].GaugeId < distRecords[j].GaugeId
+				})
+				
+				// update records and ensure that non-perpetuals pot cannot get rewards.
+				err = keeper.UpdateDistrRecords(suite.Ctx, tc.distRecords...)
+				if tc.expectedErr {
+					suite.Require().Error(err)
+				} else {
+					suite.Require().NoError(err)
+				}
+			}
+			res, err := queryClient.IncentivizedPools(context.Background(), &types.QueryIncentivizedPoolsRequest{})
+			if !tc.expectedErr {
```

Comment:
It looks like we aren't requiring errors for expectedErr = true cases right?

Reply:
yeah we should add the case fpr Require Error